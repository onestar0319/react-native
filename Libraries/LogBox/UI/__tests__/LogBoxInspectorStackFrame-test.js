/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const LogBoxInspectorStackFrame = require('../LogBoxInspectorStackFrame')
  .default;
const render = require('../../../../jest/renderer');

describe('LogBoxInspectorStackFrame', () => {
  it('should render stack frame', () => {
    const output = render.shallowRender(
      <LogBoxInspectorStackFrame
        onPress={() => {}}
        frame={{
          column: 1,
          file: 'app.js',
          lineNumber: 1,
          methodName: 'foo',
          collapse: false,
        }}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render stack frame without press feedback', () => {
    const output = render.shallowRender(
      <LogBoxInspectorStackFrame
        frame={{
          column: 1,
          file: 'app.js',
          lineNumber: 1,
          methodName: 'foo',
          collapse: false,
        }}
      />,
    );

    // Both button backgrounds should be transparent
    expect(output).toMatchSnapshot();
  });

  it('should render collapsed stack frame with dimmed text', () => {
    const output = render.shallowRender(
      <LogBoxInspectorStackFrame
        onPress={() => {}}
        frame={{
          column: 1,
          file: 'app.js',
          lineNumber: 1,
          methodName: 'foo',
          collapse: true,
        }}
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
