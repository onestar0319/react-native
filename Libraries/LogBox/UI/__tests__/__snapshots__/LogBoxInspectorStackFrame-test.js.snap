// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorStackFrame should render collapsed stack frame with dimmed text 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 15,
    }
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "transparent",
        "pressed": "rgba(51, 51, 51, 1)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "borderRadius": 5,
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Menlo",
            "fontSize": 14,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 18,
          },
          Object {
            "color": "rgba(255, 255, 255, 0.4)",
            "fontWeight": "300",
          },
        ]
      }
    >
      foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="middle"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
              "paddingLeft": 10,
            },
            Object {
              "color": "rgba(255, 255, 255, 0.4)",
              "fontWeight": "300",
            },
          ]
        }
      >
        app.js
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
            },
            Object {
              "color": "rgba(255, 255, 255, 0.4)",
              "fontWeight": "300",
            },
          ]
        }
      >
        :
        1
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
            },
            Object {
              "color": "rgba(255, 255, 255, 0.4)",
              "fontWeight": "300",
            },
          ]
        }
      >
        :
        2
      </Text>
    </View>
  </LogBoxButton>
</View>
`;

exports[`LogBoxInspectorStackFrame should render stack frame 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "paddingHorizontal": 15,
    }
  }
>
  <LogBoxButton
    backgroundColor={
      Object {
        "default": "transparent",
        "pressed": "rgba(51, 51, 51, 1)",
      }
    }
    onPress={[Function]}
    style={
      Object {
        "borderRadius": 5,
        "flex": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 4,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "fontFamily": "Menlo",
            "fontSize": 14,
            "fontWeight": "400",
            "includeFontPadding": false,
            "lineHeight": 18,
          },
          false,
        ]
      }
    >
      foo
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        ellipsizeMode="middle"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
              "paddingLeft": 10,
            },
            false,
          ]
        }
      >
        app.js
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
            },
            false,
          ]
        }
      >
        :
        1
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(255, 255, 255, 0.8)",
              "fontSize": 12,
              "fontWeight": "300",
              "includeFontPadding": false,
              "lineHeight": 16,
            },
            false,
          ]
        }
      >
        :
        2
      </Text>
    </View>
  </LogBoxButton>
</View>
`;
